df.viewscreen.xml

```diff
@@ -245,16 +223,17 @@
         <uint32_t name='flag'/>
     </class-type>
 
-    <class-type type-name='text_truncated' original-name='widgets::text_truncated' inherits-from='widget_text'>
+    <class-type type-name='widget_text_truncated' original-name='widgets::text_truncated' inherits-from='widget_text'>
         <stl-string name='original_str'/>
-        <int32_t name='truncate_mode_flags'/>
+        <bitfield name='truncate_mode_flags' base-type='int32_t'> bay12: TRUNCATE_MODE_*
+            <flag-bit name='abbreviate'/>
+            <flag-bit name='ellipses'/>
+        </bitfield>
     </class-type>
 
-    <class-type type-name='widget_text_multiline' original-name='widgets::text_multiline' inherits-from='widget_text'>
-        <stl-vector type-name='stl-string' name='strs'/>
-    </class-type>
+    <class-type type-name='widget_text_multiline' original-name='widgets::text_multiline' inherits-from='widget_text'/>
```

Comment:
this is correct because the strs vector is static... but how can widget state possibly be static like that?